### PR TITLE
Add unwinding info for all ISRs

### PR DESCRIPTION
**Pull request by gsilvis**
Use .cfi to make it actually fairly readable... at least, if you know what any
of the cfi commands are.

### CODE REVIEW

File: kernel/unwind.c [hunk 1]
```diff
@@ -333,26 +338,38 @@ static int run_dwarf_op(head *in, const eh_cie *cie, cfa_row *row) {
...
+         * unlikely that we need the information anyways. */
+        return 0;
+      }
+      row->registers[reg].rule = DW_OFFSET;
+      row->registers[reg].offset = soff * cie->data_af;
+      return 0;
     default:
       com_printf("BUG: unrecognized DWARF opcode %d\n", c);
-      return -1;
+      return 1;
```
Comment (gsilvis) [inline kernel/unwind.c]: Oops, this should still be -1.  Will fix.

**Comment by gsilvis** [comment]
Hand-tested it at -O0, -O1, -O2, -O3, -Os, and -O3 -flto, and it worked on all of them.

File: kernel/unwind.c [hunk 2]
```diff
@@ -505,7 +522,11 @@ static int find_fde(uint64_t rip, eh_cie* cie_out, eh_fde* fde_out) {
     memset(&fde, 0, sizeof(eh_fde));
     fde.cie = &current_cie;
     if (read_fde(&subhead, &fde)) { return -1; }
-    if (rip >= fde.pc_begin && rip - fde.pc_begin < fde.pc_len) {
+    /* Offset our check by one instruction: it's unlikely that we're returning
```
Comment (cstanfill) [inline kernel/unwind.c]: As we discussed, this logic is still slightly wrong for interrupt handlers. We can fault on the first instruction in a function.
Luckily for us, interrupt handlers are never noreturn, in a sense - with the exception of double-fault and exit(), they are /always/ called with the intention of actually returning to the instruction on the stack. So if we can figure out how to communicate that info into the eh_frame section, the interrupt handlers can use the more sane looking FDE check and not worry about falling off the end of any functions.

**Comment by gsilvis** [comment]
Superseded by #88

**Pull request closed without merging**